src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -851,6 +851,9 @@ async def __run_task_function(self) -> None:
             if not crawling_context.session:
                 raise RuntimeError('SessionError raised in a crawling context without a session') from session_error
 
+            if self._error_handler:
+                await self._error_handler(crawling_context, session_error)
```

Comment:
Now that I think of it, for regular (non-SessionError) failures we also handle the case when this throws as well.